sql/core/src/main/scala/org/apache/spark/sql/sources/ddl.scala

```diff
@@ -304,6 +304,9 @@ private[sql] object ResolvedDataSource {
       mode: SaveMode,
       options: Map[String, String],
       data: DataFrame): ResolvedDataSource = {
+    if (data.schema.map(_.dataType).exists(_.isInstanceOf[IntervalType])) {
+      sys.error("Cannot save interval data type into external storage.")
```

Comment:
when is this called? if it is called during analysis, it'd make more sense to throw AnalysisException, since that has better error reporting in Python.